torchao/quantization/prototype/qat.py

```diff
@@ -205,13 +205,14 @@ class _GenericFakeQuantize(torch.autograd.Function):
 
     @staticmethod
     def forward(ctx, input, scales, zero_points, quant_min, quant_max):
+        scales = scales.to(torch.float32)
+        zero_points = zero_points.to(torch.int32)
```

Comment:
what are the original inputs? why do we need to do casting here instead of when these are produced?

Reply:
These are the outputs of `choose_qparams_per_token_asymmetric` and `get_group_qparams_symmetric`. We could move the casts outside and turn these into asserts, but I think we need int32 zero points to match the numerics of `torch.fake_quantize_per_channel_affine`

Reply:
oh I see, this is supported in the new primitive: https://github.com/pytorch/ao/blob/main/torchao/quantization/quant_primitives.py#L257-L258

Reply:
yeah we should expose it in `get_group_qparams_symmetric`

Reply:
added a TODO for now